packages/workspace/src/common/workspace-file-service.ts

```diff
@@ -0,0 +1,63 @@
+// *****************************************************************************
+// Copyright (C) 2023 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { URI } from '@theia/core';
+import { injectable } from '@theia/core/shared/inversify';
+import { FileStat } from '@theia/filesystem/lib/common/files';
+
+export interface WorkspaceFileType {
+    extension: string
+    name: string
+}
+
+/**
+ * @deprecated Since 1.38.0. Use `WorkspaceFileService#getWorkspaceFileTypes` instead.
+ */
+export const THEIA_EXT = 'theia-workspace';
+export const VSCODE_EXT = 'code-workspace';
+
+@injectable()
+export class WorkspaceFileService {
+
+    /**
+     * Check if the file should be considered as a workspace file.
+     *
+     * Example: We should not try to read the contents of an .exe file.
+     */
+    isWorkspaceFile(candidate: FileStat | URI): boolean {
+        const uri = FileStat.is(candidate) ? candidate.resource : candidate;
+        const extensions = this.getWorkspaceFileTypes().map(type => `.${type.extension}`);
+        return extensions.includes(uri.path.ext);
+    }
+
+    getWorkspaceFileTypes(): WorkspaceFileType[] {
```

Comment:
The fact that the first returned file type is used as the default file type is not apparent, here, an adopter would have to look at the usages to discover that fact. I would either document the facts that 
1. You need to return at least one file type
2. The first file type is considered the "default file type".

or else add a property/method `defaultFileType` or `defaultFileTypeIndex` in order to encapsulate the convention in this class.

Reply:
Makes sense, thanks for the input :)